addons/stock/models/stock_move.py

```diff
@@ -1844,6 +1844,10 @@ def _set_quantities_to_reservation(self):
                     continue
                 move_line.qty_done = move_line.product_uom_qty
 
+    def _clear_quantities_to_zero(self):
+        self.filtered(lambda m: m.state in ('partially_available', 'assigned') and \
+            float_compare(m.quantity_done, m.reserved_availability, precision_rounding=m.product_uom.rounding) == 0).move_line_ids.qty_done = 0
```

Comment:
I think you don't need to check if `quantity_done` is equal to `reserved_availability` as you already did it in the picking's method
```suggestion
        self.filtered(lambda m: m.state in ('partially_available', 'assigned')).move_line_ids.qty_done = 0
```

Or if you want to keep the condition here, remove the filter in the picking's method as I suggested in a previous comment

Reply:
you are right, removed the check